empress/support_files/js/colorer.js

```diff
@@ -118,6 +123,10 @@ define(["chroma", "underscore", "util"], function (chroma, _, util) {
         } else {
             palette = chroma.brewer[this.color];
         }
+        if (this.reverse) {
+            palette = [...palette];
```

Comment:
I'm guessing this line is designed to convert the palette into a JS array so `.reverse()` can be used. If so, I think it might be nice to restate this [using underscore.js](https://underscorejs.org/#toArray) as

```suggestion
            palette = _.toArray(palette);
```

I doubt that this will make a difference efficiency-wise, but I think it should be more clear to readers.

Reply:
That makes sense, thanks for clarifying your intent! I think your understanding is correct; JS is a bit funky but Objects (including Arrays) are passed by reference, so it'd be possible for the `reverse` to impact the palette. Thanks for being so cautious about this.

Ok, in this case (sorry to ask for another change here) my previous suggestion was suboptimal -- I don't think `_.toArray()` makes the most sense for creating a copy, at least from the perspective of readability. Would it be possible to just replace `_.toArray()` with `_.clone()`? This way it is as clear as possible to readers _why_ this step is being taken. Alternately, I also wouldn't mind going back to the `...` syntax instead, provided we add a comment just linking to this thread or whatever (basically just explaining that this is a necessary step, in the hopes that one of us doesn't come back to this 10 years from now and go "hmm that line probably wasn't doing anything important anyway" ;).

![image](https://user-images.githubusercontent.com/4177727/96493704-29759500-11fa-11eb-8948-e1c5dc9fb831.png)

Reply:
Yeah the `_.toArray -> _.clone` change makes sense.